### PR TITLE
fix a test on the mac

### LINKED ISSUE #1709: pub run test fails in packages/flutter_tools in test/create_test.dart on Mac
Only seems to happen on Mac.  Both on the bot and on my local machine.  Travis doesn't run our tests on Mac, hence we didn't catch this prior.

```
00:01 +22 -1: test/create_test.dart: create flutter-simple                                         
  Expected: <0>
    Actual: <69>
  
  package:test                 expect
  test/create_test.dart 38:33  defineTests.<fn>.<fn>.<async>.<fn>
  ===== asynchronous gap ===========================
  dart:async                   _Future.then
  test/create_test.dart 38:14  defineTests.<fn>.<fn>.<async>
  ===== asynchronous gap ===========================
  dart:async                   Future.Future.microtask
  test/create_test.dart        defineTests.<fn>.<fn>
  
00:01 +22 -1: Some tests failed. 
```

From a quick look at the code, looks like probably the pub get its running s returning 69?  I don't see how else it could be getting 69 from that code.

This is non-critical, I can just disable this test on Mac.

FYI @devoncarew

### PR DESCRIPTION
**Pull request by devoncarew**
Fix a test failure on the mac (fix https://github.com/flutter/flutter/issues/1709). `path.relative` was not creating correct relative paths. We work around it by using absolute paths if there is no good relative path between two directories (if the relative path would go all the way up to the filesystem root and back down).

@eseidelGoogle

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/create.dart [hunk 1]
```diff
@@ -135,6 +141,16 @@ abstract class Template {
     });
   }
 
+  bool _hasRelativePath(String path1, String path2) {
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/create.dart]: I don't understand this code.
Suppose the two paths are "/foo" and "/bar/baz". There's a simple relative path from one to the other: "bar/baz". But this function will return false unless I'm misunderstanding it.

I suggest exposing this function to tests and testing it with a bunch of cases.

File: packages/flutter_tools/test/create_test.dart [hunk 2]
```diff
@@ -25,34 +25,28 @@ defineTests() {
...
+      CreateCommand command = new CreateCommand();
+      CommandRunner runner = new CommandRunner('test_flutter', '')
+        ..addCommand(command);
+      await runner.run(['create', '--out', temp.path])
+          .then((int code) => expect(code, equals(0)));
+
+      String mainPath = path.join(temp.path, 'lib', 'main.dart');
+      expect(new File(mainPath).existsSync(), true);
+      ProcessResult exec = Process.runSync(
+        sdkBinaryName('dartanalyzer'), ['--fatal-warnings', mainPath],
```
Comment (Hixie) [inline packages/flutter_tools/test/create_test.dart]: This is an odd thing for us to do. Shouldn't we use flutter analyze?

Comment (devoncarew) [inline packages/flutter_tools/test/create_test.dart]: This pre-dates flutter analyze. We probably want to switch it over.

File: packages/flutter_tools/lib/src/commands/create.dart (hunk 1 above)
Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/create.dart]: I'll add some tests. This method is indented to work with two absolute paths. When split, the paths parts will be `/`, `foo`, ... and `/`, `bar`, ... . I'm trying to verify that the first non-slash part of the paths are the same.

**Comment by devoncarew** [comment]
PTAL, re-worked to remove the `_hasRelativePath` method. We now validate the the relative path that `path.relative` created exists; if not, we fall back to using an absolute path.

**Comment by Hixie** [comment]
When will the path not exist?

**Comment by devoncarew** [comment]
I think `path.relative` is buggy. When this code was refactored to put relative paths in the pubspec, on the mac, for this test, one of the paths is the /var temp dir and the relative path produced for flutter_root/packages/flutter from the temp path is incorrect.

**Comment by Hixie** [comment]
We should file a bug upstream in that case.

**Comment by chinmaygarde** [comment]
Is this PR still open? Trying to close some tickets :)

**Comment by devoncarew** [comment]
Yeah, I haven't filed an upstream bug. Need to get some repro steps. We could close until that case.

**Pull request closed without merging**